src/DynamoCoreWpf/Views/Search/LibrarySearchView.xaml.cs

```diff
@@ -177,7 +183,20 @@ private void ShowTooltip(object sender)
             if (fromSender == null) return;
 
             libraryToolTipPopup.PlacementTarget = fromSender;
-            libraryToolTipPopup.SetDataContext(fromSender.DataContext);
+            if ((fromSender.DataContext as NodeSearchElementViewModel).Visibility)
+                libraryToolTipPopup.SetDataContext(fromSender.DataContext);
+            else
+                CloseTooltip();
+        }
+
+        private void CloseTooltip(bool closeImmediately = false)
```

Comment:
I was a little confused by method `CloseTooltip` and event handler `CloseTooltip`, both are having the same name. If I may suggest, perhaps this method can be named `CloseToolTipInternal` and the event handler below can be named `OnRequestCloseToolTip`? (Note that to align with the control name, please use `ToolTip` rather than `Tooltip`.)